### ISSUE TITLE
False Positive On Aliases In Mapping Nodes

### ISSUE THREAD

**Issue by Kurt-von-Laven** [rhysd/actionlint#133]
Actionlint complains `"with" section is alias node but mapping node is expected [syntax-check]` on valid YAML syntax when anchors and aliases are used with mapping nodes:

```yaml
- name: Run my-action.
  run: my-organization/my-action
  with: &my-action-inputs
    a: b
    c: d
    e: f
- name: Run my-action again.
  run: my-organization/my-action
  with: *my-action-inputs
```

I am uncertain how one would suppress the error.

**Comment by rhysd** [comment]
Yeah, aliases are actually not considered because I've never used them in my workflows and anyone didn't report that previously. But I agree this should not cause false positive at least.

**Comment by rhysd** [comment]
I created a minimal workflow to reproduce this:

```yaml
name: Test
on: push

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - name: Run my-action.
        uses: my-organization/my-action@main
        with: &my-action-inputs
          a: b
          c: d
          e: f
      - name: Run my-action again.
        uses: my-organization/my-action@main
        with: *my-action-inputs
```

**Comment by prein** [comment]
Consider also yaml anchors (if you didn't already)
Example:
```
name:  Test
on: push
jobs:
  test1:
    runs-on: ubuntu-latest
    steps:
    - &run_action
      name: Run my-action
      uses: my-organization/my-action@main
  test2:
    runs-on: ubuntu-latest
    steps:
    - name: Restore built artifacts from cache
      <<: *run_action
```

**Comment by Kurt-von-Laven** [comment]
This doesn't solve your problem, but to toot my own horn, [MegaLinter](https://megalinter.io) runs significantly faster than Super-Linter, and has many more linters. It also has actionlint.

**Comment by muzimuzhi** [comment]
GitHub Actions has just supported YAML anchors in workflow files (verified myself) and perhaps also in action files (not verified). So this issue now becomes an indeed enhancement issue.

See https://github.com/actions/runner/issues/1182.
